kafka-client/src/test/java/dev/responsive/kafka/internal/db/mongo/MongoSessionTableTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.responsive.kafka.internal.db.mongo;
+
+import static dev.responsive.kafka.api.config.ResponsiveConfig.STORAGE_HOSTNAME_CONFIG;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.mongodb.client.MongoClient;
+import dev.responsive.kafka.api.config.StorageBackend;
+import dev.responsive.kafka.internal.db.partitioning.SessionSegmentPartitioner;
+import dev.responsive.kafka.internal.utils.SessionKey;
+import dev.responsive.kafka.internal.utils.SessionUtil;
+import dev.responsive.kafka.testutils.ResponsiveConfigParam;
+import dev.responsive.kafka.testutils.ResponsiveExtension;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.KeyValue;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class MongoSessionTableTest {
+
+  @RegisterExtension
+  public static final ResponsiveExtension EXT = new ResponsiveExtension(StorageBackend.MONGO_DB);
+  private static final CollectionCreationOptions UNSHARDED = new CollectionCreationOptions(
+      false,
+      0
+  );
+  private static final byte[] DEFAULT_VALUE = new byte[] {1};
+
+  private String name;
+  private MongoClient client;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      @ResponsiveConfigParam final Map<String, Object> props
+  ) {
+    name = info.getDisplayName().replace("()", "");
+
+    final String mongoConnection = (String) props.get(STORAGE_HOSTNAME_CONFIG);
+    client = SessionUtil.connect(mongoConnection, null, null);
+  }
+
+  @Test
+  public void shouldSucceedSimpleSetGet() {
```

Comment:
are there any meaningful negative tests we should test? (i.e. throwing exceptions)

also suggestion - we should consider splitting out: `shouldFindInsertedKeyWithValidStartTimestamp`, `shouldNotFindInsertedKeyWithInvalidStartTimestamp` and `shouldNotFindKeyThatWasNotInserted`

That way it's very clear what each of the tests are doing and when they fail the test name says it all

Reply:
No negative test that I can think of right now, I wrote those tests as mirrors of the ones for the KV and Window stores but we might want to expand all of them in similar ways (to include negative tests as you mentioned).

Reply:
Expanding the basic test coverage (ie unit tests) of all the stores has been on my TODO list for a while. It's hard to make time to prioritize writing tests for things that seem to be working when there is so much else to do -- but we don't want to wait until it's too late to catch bugs. 

Perhaps we could dedicate something like one day each month towards "bug bashing and bug prevention" -- including filling in holes in our test coverage. I think it would be pretty valuable. Even a half day a month could pay off if all of us do it together